ci/jwt/jwks_server.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+
+import argparse
+import base64
+import json
+import logging
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from collections import OrderedDict
+from jwcrypto import jwt, jwk
+
+DEFAULT_PORT = 8080
+
+keys = {}
+
+
+class JWKSRequestHandler(BaseHTTPRequestHandler):
+
+    def reply(self, response):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+        self.wfile.write(response.encode('utf-8'))
+
+    def get_key(self, key_name, alg):
+        if key_name not in keys:
+            keys[key_name] = {
+                'RS256': jwk.JWK.generate(kty='RSA', size=1024),
+                'RS384': jwk.JWK.generate(kty='RSA', size=2048),
+                'RS512': jwk.JWK.generate(kty='RSA', size=4048),
+                'ES256': jwk.JWK.generate(kty='EC', crv='P-256'),
+                'ES384': jwk.JWK.generate(kty='EC', crv='P-384'),
+                'ES512': jwk.JWK.generate(kty='EC', crv='P-521'),
+                'HS256': jwk.JWK.generate(kty='oct', size=256),
+                'HS384': jwk.JWK.generate(kty='oct', size=384),
+                'HS512': jwk.JWK.generate(kty='oct', size=512)
+            }.get(alg, None)
+        return keys[key_name]
+
+    def export_key(self, key):
+        try:
+            key_json = key.export(private_key=False)
+        except jwk.InvalidJWKType:
+            key_json = key.export()
+        key_dict = json.loads(key_json, object_pairs_hook=OrderedDict)
+        key_dict['kid'] = key.thumbprint()
+        return key_dict
+
+    def base64_padding(self, value):
+        return value + "=" * (-len(value) % 4)
+
+    def decode_token(self, token):
+        head_b64, payl_b64, sig = token.split(".", 3)
+        head = base64.urlsafe_b64decode(self.base64_padding(head_b64))
+        payl = base64.urlsafe_b64decode(self.base64_padding(payl_b64))
+        head_dict = json.loads(head, object_pairs_hook=OrderedDict)
+        payl_dict = json.loads(payl, object_pairs_hook=OrderedDict)
+        return head_dict, payl_dict
+
+    def do_GET(self): # pylint: disable=C0103
```

Comment:
do_GET and do_POST seems to do many things in each function and can be broken to smaller more modular functions

Reply:
Add docstring to this functions about what they do

Reply:
Regarding do_GET I'm agree, but do_POST does exactly one thing...